### PR TITLE
Patches to address static analysis warnings

### PR DESCRIPTION
**Pull request by MisterDA**
Hi!
This is a collection of small patches fixing some warnings reported by PVS-Studio.
Happy to change things if I got the "bug fix" wrong.

I’m not so sure about my fix for 7d1d619 "types/wlr_output: fix pointer to stack variable escaping scope".

### CODE REVIEW

File: xwayland/xwayland.c [hunk 1]
```diff
@@ -91,7 +91,6 @@ struct wlr_xwayland *wlr_xwayland_create(struct wl_display *wl_display,
 	};
 	xwayland->server = wlr_xwayland_server_create(wl_display, &options);
 	if (xwayland->server == NULL) {
-		free(xwayland->server);
```
Comment (emersion) [inline xwayland/xwayland.c:94]: Typo: this should free `xwayland` instead

File: render/drm_format_set.c [hunk 2]
```diff
@@ -76,15 +76,16 @@ bool wlr_drm_format_set_add(struct wlr_drm_format_set *set, uint32_t format,
 
 		if (fmt->len == fmt->cap) {
 			size_t cap = fmt->cap ? fmt->cap * 2 : 4;
-
-			fmt = realloc(fmt, sizeof(*fmt) + sizeof(fmt->modifiers[0]) * cap);
```
Comment (emersion) [inline render/drm_format_set.c:80]: I don't understand why this change is necessary.

Comment (MisterDA) [reply render/drm_format_set.c:80]: `fmt` leaks memory if the reallocation fails, because it will be overwritten with `NULL`. This patch ensures that we keep the old data in place if the reallocation fails.

Comment (emersion) [reply render/drm_format_set.c:80]: `fmt` is a local variable, so it doesn't matter right?

Comment (MisterDA) [reply render/drm_format_set.c:80]: You’re right.
